src/gpu/align_z_to_other.cu

```diff
@@ -0,0 +1,208 @@
+
+        if (depth_x >= depth_intrin.width || depth_y >= depth_intrin.height) {
+            return;
+        }
+
+        int depth_pixel_index = depth_y * depth_intrin.width + depth_x;
+
+        // Skip over depth pixels with the value of zero, we have no depth data so we will not write anything into our aligned images
+        if (float depth = get_depth(depth_pixel_index, z_scale, z_pixels_gpu)) {
+            // Map the top-left corner of the depth pixel onto the other image
+            float depth_pixel[2] = {depth_x - 0.5f,
+                                    depth_y - 0.5f}, depth_point[3], other_point[3], other_pixel[2];
+            rs_deproject_pixel_to_point(depth_point, &depth_intrin, depth_pixel, depth, inv_fx, inv_fy);
+            rs_transform_point_to_point(other_point, &depth_to_other, depth_point);
+            rs_project_point_to_pixel(other_pixel, &other_intrin, other_point);
+            const int other_x0 = static_cast<int>(other_pixel[0] + 0.5f);
+            const int other_y0 = static_cast<int>(other_pixel[1] + 0.5f);
+
+            // Map the bottom-right corner of the depth pixel onto the other image
+            depth_pixel[0] = depth_x + 0.5f;
+            depth_pixel[1] = depth_y + 0.5f;
+            rs_deproject_pixel_to_point(depth_point, &depth_intrin, depth_pixel, depth, inv_fx, inv_fy);
+            rs_transform_point_to_point(other_point, &depth_to_other, depth_point);
+            rs_project_point_to_pixel(other_pixel, &other_intrin, other_point);
+            const int other_x1 = static_cast<int>(other_pixel[0] + 0.5f);
+            const int other_y1 = static_cast<int>(other_pixel[1] + 0.5f);
+
+            if (other_x0 < 0 || other_y0 < 0 || other_x1 >= other_intrin.width ||
+                other_y1 >= other_intrin.height)
+                return;
+            // Transfer between the depth pixels and the pixels inside the rectangle on the other image
+            for (int y = other_y0; y <= other_y1; ++y)
+                for (int x = other_x0; x <= other_x1; ++x)
+                    transfer_pixel(depth_pixel_index, y * other_intrin.width + x, z_pixels_gpu, out_pixels_gpu);
+        }
+    }
+
+
+    class CudaStreamWrapper {
+    private:
+        cudaStream_t stream;
+    public:
+        CudaStreamWrapper() {
+            cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking);
+            ROS_INFO_THROTTLE(60, "GPU Stream Created");
+
+        }
+
+        ~CudaStreamWrapper() {
+            cudaStreamDestroy(stream);
+            ROS_INFO_THROTTLE(60, "GPU Stream Destroyed");
+
+        }
+
+        operator cudaStream_t&(){
+            return stream;
+        }
+
+
+    };
+
+    __host__ bool align_z_to_other(rsimpl::byte * z_aligned_to_other, const uint16_t * z_pixels, float z_scale, const rs_intrinsics & z_intrin, const rs_extrinsics & z_to_other, const rs_intrinsics & other_intrin)
+    {
+        ROS_INFO_THROTTLE(60, "GPU NOT DISABLED");
```

Comment:
Should we just print INFO level messages once per run?

Reply:
this is for debugging... not sure if we want to spam the logs